### PR TITLE
feat: yearn multi account

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This adds multi account support for Yearn DeFi opportunities:

- Overview
- Deposit
- Withdraw

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/2605

## Risk

Yearn modal should be fully regression tested - in theory, this brings no changes for now and uses programmatic accountNumber vs. hardcoded accountNumber, but this requires a breaking(ish) change in the lib PR.

## Testing

- With the `MultiAccounts` flag on, the account dropdown should be present in Yearn opportunities Overview, Deposit, and Withdraw steps
- With the `MultiAccounts` flag off, there should be no user-facing changes nor regressions with the current flow

### Engineering

- Refer to the top-level testing notes
- `console.log` or debug `accountNumber` props in components to ensure the right account number is passed down

### Operations

- Refer to the top-level testing notes
- Regression test the whole Yearn modal flow, including making deposit and withdraws

## Screenshots (if applicable)

image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
          * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

File: src/features/defi/providers/yearn/components/YearnManager/Deposit/YearnDeposit.tsx [hunk 1]
```diff
@@ -95,12 +99,15 @@ export const YearnDeposit = () => {
...
+  const filter = useMemo(() => ({ accountId: accountId ?? '' }), [accountId])
+  const accountNumber = useAppSelector(state => selectAccountNumberByAccountId(state, filter))
+
   const StepConfig: DefiStepProps = useMemo(() => {
     return {
       [DefiStep.Info]: {
         label: translate('defi.steps.deposit.info.title'),
         description: translate('defi.steps.deposit.info.description', { asset: asset.symbol }),
-        component: Deposit,
+        component: props => <Deposit {...props} onAccountChange={handleAccountChange} />,
```
Comment (gomesalexandre) [inline src/features/defi/providers/yearn/components/YearnManager/Deposit/YearnDeposit.tsx:110]:
```suggestion
        component: ownProps => <Deposit {...ownProps} onAccountChange={handleAccountChange} />,
```

File: src/features/defi/providers/yearn/components/YearnManager/Deposit/YearnDeposit.tsx [hunk 2]
```diff
@@ -111,14 +118,14 @@ export const YearnDeposit = () => {
       },
       [DefiStep.Confirm]: {
         label: translate('defi.steps.confirm.title'),
-        component: Confirm,
+        component: props => <Confirm {...props} accountNumber={accountNumber} />,
```
Comment (gomesalexandre) [inline src/features/defi/providers/yearn/components/YearnManager/Deposit/YearnDeposit.tsx:121]:
```suggestion
        component: ownProps => <Confirm {...ownProps} accountNumber={accountNumber} />,
```

File: src/features/defi/providers/yearn/components/YearnManager/Withdraw/YearnWithdraw.tsx [hunk 3]
```diff
@@ -105,25 +109,28 @@ export const YearnWithdraw = () => {
...
+
   const StepConfig: DefiStepProps = useMemo(() => {
     return {
       [DefiStep.Info]: {
         label: translate('defi.steps.withdraw.info.title'),
         description: translate('defi.steps.withdraw.info.description', {
           asset: underlyingAsset.symbol,
         }),
-        component: Withdraw,
+        component: props => <Withdraw {...props} onAccountChange={handleAccountChange} />,
```
Comment (gomesalexandre) [inline src/features/defi/providers/yearn/components/YearnManager/Withdraw/YearnWithdraw.tsx:122]:
```suggestion
        component: ownProps => <Withdraw {...ownProps} onAccountChange={handleAccountChange} />,
```

File: src/features/defi/providers/yearn/components/YearnManager/Withdraw/YearnWithdraw.tsx [hunk 4]
```diff
@@ -105,25 +109,28 @@ export const YearnWithdraw = () => {
...
       },
       [DefiStep.Confirm]: {
         label: translate('defi.steps.confirm.title'),
-        component: Confirm,
+        component: props => <Confirm {...props} accountNumber={accountNumber} />,
```
Comment (gomesalexandre) [inline src/features/defi/providers/yearn/components/YearnManager/Withdraw/YearnWithdraw.tsx:126]:
```suggestion
        component: ownProps => <Confirm {...ownProps} accountNumber={accountNumber} />,
```

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
        * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈
          * **PR #2680** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2680" target="_blank">Graphite</a> 
          * **PR #2669** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2669" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
        * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈
          * **PR #2680** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2680" target="_blank">Graphite</a> 
          * **PR #2669** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2669" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
        * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈
          * **PR #2680** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2680" target="_blank">Graphite</a> 
          * **PR #2669** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2669" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
        * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈
          * **PR #2680** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2680" target="_blank">Graphite</a> 
          * **PR #2669** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2669" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
        * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈
          * **PR #2680** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2680" target="_blank">Graphite</a> 
          * **PR #2669** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2669" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
        * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈
          * **PR #2680** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2680" target="_blank">Graphite</a> 
          * **PR #2669** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2669" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
        * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈
          * **PR #2680** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2680" target="_blank">Graphite</a> 
          * **PR #2669** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2669" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
        * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈
          * **PR #2680** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2680" target="_blank">Graphite</a> 
          * **PR #2669** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2669" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
        * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈
          * **PR #2680** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2680" target="_blank">Graphite</a> 
          * **PR #2669** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2669" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
        * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈
          * **PR #2680** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2680" target="_blank">Graphite</a> 
          * **PR #2669** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2669" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2642** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2642" target="_blank">Graphite</a> 
    * **PR #2644** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2644" target="_blank">Graphite</a> 
      * **PR #2651** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2651" target="_blank">Graphite</a> 
        * **PR #2652** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2652" target="_blank">Graphite</a> 
        * **PR #2661** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2661" target="_blank">Graphite</a>  👈
          * **PR #2680** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2680" target="_blank">Graphite</a> 
          * **PR #2669** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2669" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/2661?utm_source=stack-comment).

**Pull request merged**